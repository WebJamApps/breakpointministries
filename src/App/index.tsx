import superagent from 'superagent';
import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { connect } from 'react-redux';
import commonUtils from '../lib/commonUtils';
import AdminDashboardDefault from '../containers/AdminDashboard';
import AppFourOhFour from './404';
import AppTemplateDefault from './AppTemplate';
import DefaultHome from '../containers/Homepage';
import mapStoreToProps from '../redux/mapStoreToProps';
import fetch from '../lib/fetch';
import { AppProps } from './AppTypes';

export class App extends Component<AppProps> {
  fetch: typeof fetch;

  superagent: superagent.SuperAgentStatic;

  static defaultProps = {
    dispatch: /* istanbul ignore next */(): void => { },
    auth: {
      isAuthenticated: false,
      user: { userType: '' },
      error: '',
      email: '',
      token: '',
    },
  };

  constructor(props: AppProps) {
    super(props);
    this.fetch = fetch;
    this.state = {};
    this.superagent = superagent;
  }

  componentDidMount(): void { // fetch the blogs to populate homepage content
    this.fetch.fetchGet(this, 'blog/', 'GOT_BLOGS');
  }

  render(): JSX.Element {
    const { auth } = this.props;
    const userRoles: string[] = commonUtils.getUserRoles();
    return (
      <div id="App" className="App">
        <Router>
          <AppTemplateDefault>
            <Switch>
              <Route exact path="/" component={DefaultHome} />
              {auth.isAuthenticated && auth.user.userType && userRoles.indexOf(auth.user.userType) !== -1
                ? <Route path="/admin" component={AdminDashboardDefault} /> : null}
              <Route component={AppFourOhFour} />
            </Switch>
          </AppTemplateDefault>
        </Router>
      </div>

    );
  }
}

export default connect(mapStoreToProps, null)(App);
