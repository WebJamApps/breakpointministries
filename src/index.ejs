<!DOCTYPE html>
<html>
  <head prefix="og: http://ogp.me/ns# fb: http://ogp.me/ns/fb#">
    <meta property="og:title" content="Change In Christ" />
    <meta property="og:description" content="">
    <meta property="og:image" content="">
    <meta property="og:url" content="https://www.changeinchrist.org">
    <title>
      <%- htmlWebpackPlugin.options.metadata.title %>
    </title>
    <base href="<%- htmlWebpackPlugin.options.metadata.baseUrl %>" />
    <link
      href="https://fonts.googleapis.com/css?family=PT Sans Caption"
      rel="stylesheet"
    />
    <link href="https://fonts.googleapis.com/css?family=Habibi" rel="stylesheet" />
    <link
      href="https://fonts.googleapis.com/icon?family=Material+Icons"
      rel="stylesheet"
    />
    <link
      rel="stylesheet"
      href="https://use.fontawesome.com/releases/v5.8.1/css/all.css"
      integrity="sha384-50oBUHEmvpQ+1lW4y57PTFmhCaXp0ML5d60M1M7uH2+nqUivzIebhndOJK28anvf"
      crossorigin="anonymous"
    />

    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
  </head>
  <body>
    <div id="root">
      <div class="splash">
        <div class="message">
          <%- htmlWebpackPlugin.options.metadata.title %>
        </div>
        <i class="fa fa-spinner fa-spin"></i>
      </div>
    </div>
  </body>
</html>
