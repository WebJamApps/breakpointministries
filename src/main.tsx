import React from 'react';
import { Provider } from 'react-redux';
import { render } from 'react-dom';
import { PersistGate } from 'redux-persist/integration/react';
import ConnectedApp from './App';
import store from './redux/store';
import '../static/main.scss';

export const MainStack = () => (
  <Provider store={store.store}>
  <PersistGate loading={null} persistor={store.persistor}>
    <ConnectedApp />
  </PersistGate>
</Provider>
);

export const renderMain = ():void => {
  render(
<MainStack/>, document.getElementById('root'),
  );
  /* istanbul ignore next */
  if (process.env.NODE_ENV === 'development' && module.hot) module.hot.accept();
};
