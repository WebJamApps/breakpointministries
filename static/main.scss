@import 'abstracts/mixins';
@import 'abstracts/variables';

@import 'base/animations';
@import 'base/base';
@import 'base/typography';

@import 'components/button';
@import 'components/form';
@import 'components/sidebar';

@import 'layout/blog';
@import 'layout/footer';
@import 'layout/navigation';

@import 'pages/admin';
