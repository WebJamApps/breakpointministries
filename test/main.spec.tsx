
/* eslint-disable @typescript-eslint/no-explicit-any */
import React from 'react';
import renderer, { act } from 'react-test-renderer';
import { MainStack } from '../src/main';

describe('Main', () => {
  it('renders correctly', () => {
    act(() => {
      const mainStack = renderer.create(<MainStack />).toJSON();
      expect(mainStack).toMatchSnapshot();
    });
  });
});
