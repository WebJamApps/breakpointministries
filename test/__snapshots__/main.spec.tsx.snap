// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders correctly 1`] = `
<div
  className="App"
  id="App"
>
  <div
    className="open container"
  >
    <div
      className="close sidebar"
    >
      <div
        className="sidebar__content"
      >
        <header
          className="header"
        >
          <div
            className="header__logo"
          >
            <img
              alt="The Jesus Fish with a multi-coloured background"
              className="header__logo--picture"
              src="https://dl.dropbox.com/s/befh330hypey44l/fish%20with%20color.png?dl=0"
            />
            <span
              className="close header__mobile-menu"
              onClick={[Function]}
              onKeyPress={[Function]}
              role="button"
              tabIndex={0}
            >
              <span
                className="header__mobile-menu--icon"
              />
            </span>
            <span
              className="header__title"
            >
              <h2>
                Change In Christ
              </h2>
            </span>
          </div>
        </header>
        <div
          className="close sidebar__functions"
        >
          <div
            className="search"
          >
            <h2
              className="search__heading heading-2"
            >
              Search
            </h2>
            <form
              action="get"
              className="search__form"
            >
              <input
                className="search__form--input"
                defaultValue=""
                placeholder="Search..."
                role="searchbox"
                type="text"
              />
              <input
                className="btn search__form--btn"
                type="submit"
                value="Search"
              />
            </form>
          </div>
          <div
            className="subscribe"
          >
            <h2
              className="subscribe__heading heading-2"
            >
              Subscribe
            </h2>
            <form
              action="get"
              className="subscribe__form"
            >
              <input
                className="subscribe__form--input"
                defaultValue=""
                placeholder="Subscribe..."
                role="searchbox"
                type="text"
              />
              <input
                className="btn subscribe__form--btn"
                type="submit"
                value="Subscribe"
              />
            </form>
          </div>
        </div>
        <div
          className="navigation close"
        >
          <div
            className="nav"
          >
            <a
              className="nav__link"
              href="/"
              onClick={[Function]}
            >
              <i
                className="fas fa-home"
              />
              Â 
              <span
                className="nav-item"
              >
                Home
              </span>
            </a>
          </div>
          <div
            className="menu-item googleLogin"
          >
            <button
              className="googleLoginButton"
              disabled={true}
              onClick={[Function]}
              onMouseDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#fff",
                  "border": "1px solid transparent",
                  "borderRadius": 2,
                  "boxShadow": "0 2px 2px 0 rgba(0, 0, 0, .24), 0 0 1px 0 rgba(0, 0, 0, .24)",
                  "color": "rgba(0, 0, 0, .54)",
                  "display": "inline-flex",
                  "fontFamily": "Roboto, sans-serif",
                  "fontSize": 14,
                  "fontWeight": "500",
                  "opacity": 0.6,
                  "padding": 0,
                }
              }
              type="button"
            >
              <span
                style={
                  Object {
                    "fontWeight": 500,
                    "paddingBottom": 10,
                    "paddingLeft": 10,
                    "paddingRight": 10,
                    "paddingTop": 10,
                  }
                }
              >
                <i
                  className="fab fa-google"
                >
                   Login
                </i>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
    <div
      className="blog-container"
    >
      <div
        className="blog"
      >
        <div
          className="blog__entry"
        >
          <section
            className="blog__entry--body"
          >
            <p
              className="blog__entry--no-blogs"
            >
              There are no blog entries at this time.
            </p>
          </section>
        </div>
      </div>
    </div>
    <div
      className="footer"
    >
      <div
        className="footer__poweredBy"
      >
        <p>
          Powered By
           
          <a
            href="https://web-jam.com/"
            rel="noopener noreferrer"
            target="_blank"
          >
            <u>
              <strong>
                Web Jam LLC
              </strong>
            </u>
          </a>
        </p>
      </div>
    </div>
  </div>
</div>
`;
